sb.c

```diff
@@ -103,17 +103,17 @@ int sb_appendf(StringBuilder *sb, const char *format, ...)
 
 /*
  * sb_concat returns a concatenation of strings that have been appended to the
- * StringBuilder. It is the callers responsibility to free the returned
+ * stringbuilder. It is the callers responsibility to free the returned
  * reference.
  *
- * The StringBuilder is not modified by this function and can therefore continue
+ * The stringbuilder is not modified by this function and can therefore continue
  * to be used.
  */
-char *sb_concat(StringBuilder *sb)
+char *sb_concat(stringbuilder *sb)
 {
 	char			*buf = NULL;
 	char			*c = NULL;
-	StringFragment	*frag = NULL;
+	stringfragment	*frag = NULL;
 
 	buf = (char *) malloc((sb->length + 1) * sizeof(char));
```

Comment:
In read_cpuacct_usage_all() you can call sb_concat(NULL).  Here you deference sb->length without checking.

Reply:
Please squash these commits, it will make this easier to analyze.